platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/TransactionPool.java

```diff
@@ -210,8 +211,8 @@ public synchronized boolean submitTransaction(
      * Same as {@link #submitTransaction(ConsensusTransactionImpl, boolean)} but with priority set to true.
      * This method has no return since system transactions are never rejected.
      */
-    public synchronized void submitSystemTransaction(@NonNull final StateSignatureTransaction transaction) {
-        submitTransaction(transaction, true);
+    public synchronized void submitSystemTransaction(@NonNull final StateSignaturePayload transaction) {
+        submitTransaction(new StateSignatureTransaction(transaction), true);
```

Comment:
This method should probably be renamed. This method only accepts signature transactions but the name suggests this method accepts arbitrary system transactions.

Reply:
renamed